include/seqan/index/find2_index_approx.h

```diff
@@ -567,43 +724,57 @@ inline void _optimalSearchScheme(TDelegate & delegate,
                                  std::array<OptimalSearch<nbrBlocks>, N> const & ss,
                                  TDistanceTag const & /**/)
 {
+    auto dummy = [](auto /*iter*/, uint32_t /*needleLeftPos*/, uint32_t /*needleRightPos*/, uint8_t /*errors*/,
+                    auto /*s*/, uint8_t const /*blockIndex*/)
+    {
+        return(false);
+    };
+    auto delegateDummy = [](Pair<uint16_t, uint32_t> const & /*pos*/, DnaString const & /*needle*/,
+                            uint8_t const /*errors*/)
+    {};
+
     for (auto & s : ss)
-        _optimalSearchScheme(delegate, it, needle, s, TDistanceTag());
+        _optimalSearchScheme(delegate, delegateDummy, dummy, it, needle, s, TDistanceTag());
```

Comment:
* Rename `dummy` to `conditionDummy`

* Call _optimalSearchScheme(delegate, delegateDummy, conditionDummy, it, needle, ss, TDistanceTag()) to avoid code duplication (for each loop over ss)